src/shared/tpm2-util.c

```diff
@@ -1473,6 +1458,18 @@ int tpm2_digest_hash_digests(
         return tpm2_digest_hash_buffers(alg, digest, data, len, count, extend);
 }
 
+/* TPM2B_AUTH is just a typedef of TPM2B_DIGEST; let's assert that to be sure. */
+#define tpm2_digest_from_auth(auth)                                     \
```

Comment:
what good is this for?

i mean, if TPM2B_AUTH and TPM2B_DIGEST are typedef'ed to each other you can use them interchangable in a typesafe way anyway, no?

i.e.:

```c
TPM2B_AUTH *a;
TPM2B_DIGEST *b;

a = b;
```

that should not result in a compiler warning. Hence if the compiler is fine with this, then why check this explicitly another time?

Reply:
indeed, removed these and changed to passing the auth directly to the tpm2_digest functions, thanks